### PR TITLE
create oc cluster add

### PR DESCRIPTION
**Pull request by deads2k**
This gives us a starting point for `oc cluster add`.  It also removes `oc cluster up --service-catalog` and instead you can add it after the fact with `oc cluster add service-catalog` or during the command with `oc cluster up --enable=service-catalog`.

@pmorie @jpeeler fyi
/assign @mfojtik 
/assign @soltysh

### CODE REVIEW

File: pkg/oc/bootstrap/clusteradd/cmd.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	err := componentinstall.InstallComponents(componentsToInstall, c.dockerClient, c.GetLogDir())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type ClusterAddConfig struct {
+	BaseDir             string
```
[the remark this answers is not in the document]
Comment (soltysh) [reply pkg/oc/bootstrap/clusteradd/cmd.go]: > image (maybe a format here)

Definitely format, from what I remember talking to Clayton he mentioned format that we can change depending on the env.

**Comment by mfojtik** [comment]
@deads2k do we care about `oc cluster remove` ?

**Comment by deads2k** [comment]
> @deads2k do we care about oc cluster remove ?

We care, but nothing supports it yet.

**Pull request merged**